skrub/_interpolation_joiner.py

```diff
@@ -351,13 +358,14 @@ def _get_estimator_assignments(self):
         When the estimator does not handle multi-output, an estimator is fitted
         separately to each column.
         """
-        aux_table = self.aux_table.drop(self._aux_key, axis=1)
+        aux_table = std(self.aux_table).drop_columns(*self._aux_key)
+        ns = skrubns(aux_table.dataframe)
         assignments = []
-        regression_table = aux_table.select_dtypes("number")
+        regression_table = ns.select(aux_table.dataframe, ns.Selector.NUMERIC)
```

Comment:
the dataframe-api way to do this would be
```python
pdx = aux_table.__dataframe_namespace__()
aux_table.select(
    *[
        col.name
        for col in aux_table.columns_iter()
        if pdx.is_dtype(col.dtype, "numeric")
    ]
)
```